### ISSUE TITLE
Ledger support Neon3?

### ISSUE THREAD

**Issue by fabwa** [CityOfZion/neon-wallet-desktop#158]
It's not clear how to use Ledger with this Release. Any guidance?

**Comment by thiagocbalducci** [comment]
Hi @fabwa! You just connect your Ledger and open our app, once you have gone through our onboarding process and arrive at the home screen, you will be able to select your Ledger Wallet from the "Wallets" screen, there is a selector on the upper left corner of the screen (to select which wallet you want to see data from)

**Comment by Voll3r** [comment]
I have two issues that are very much related:

1. Is there any way to manually re-trigger the import of a ledger wallet after the initial onboarding? I was getting very confused trying to organise my wallets (wallets + accounts) which resulted in me removing all wallets (except the app won't allow me to delete the last one). I even uninstalled and reinstalled the app. However even the uninstall didn't reset the app. I now can't seem to import my ledger wallet manually anymore.

2. The onboarding didn't seem to include an option to import a legacy NEO wallet. I know I'm behind on this one, but since Neon seems to be able to send and receive legacy NEO just fine, it's very frustrating not to have the option to import a legacy wallet from ledger.

Bottom line: I think the app would do well to include a manual option to import a ledger wallet (both N3 and legacy). Limiting this to the onboarding process seems needlessly restrictive. Some people might want to use multiple ledgers. Some people might want to skip the onboarding (for whatever reason) and add the ledger later. Other people (like me) may not have switched to N3 just yet and might need to add both an N3 and a legacy ledger wallet.

Would love to see this improved.

Thanks!

Matt

**Comment by thiagocbalducci** [comment]
Hi @fabwa How are things going? Our revamped Ledger experience (including Login with Ledger) will be available this week, with version 3.1.0, it would be awesome to hear your feedback on it :)

**Comment by gromakovsky** [comment]
Hi @Voll3r, did you manage to find a way to migrate legacy NEO stored on a Ledger Nano S wallet to N3? Apparently the only wallet software that supports Ledger Nano S is neon-wallet, but the latest version (I've just installed 3.1.0) doesn't seem to provide migration for hardware wallets. There is "MIGRATE FROM NEON 2" button, but it seems to work only in case you know the private key/mnemonic of your old wallet, which is not the case for hardware wallets.

**Comment by Voll3r** [comment]
Hi @gromakovsky,

so far I'm waiting for the ability to access a legacy wallet via ledger to be implemented in Neon. @thiagocbalducci has indicated under issue #192 that the development team are looking into such a feature.

You could try using O3, though I haven't had any luck with it myself. It connects fine to the ledger also as a legacy wallet, but for some reason I am unable to transfer my NEO out of it. Oddly enough, transferring GAS works fine.

You might have more luck than me!

Note that for the actual conversion/migration you will not be able to do it through a wallet feature. AFAIK two possibilities exist:

1. If you have a Binance (not Binance US) account, you can receive legacy NEO in your account. After receipt, they can be withdrawn again as N3.

2. There is a smart contract somewhere that will do the conversion. Legacy NEO need to be sent to a specific address, and they will be returned to the corresponding N3 address after a couple of days.

I'll attempt the number 1 route once I have the ability to transfer out from my wallet.

**Comment by thiagocbalducci** [comment]
Hey there @gromakovsky! I will try to expand upon the aid @Voll3r gave you (thanks for helping, mate!)

Aside from the two options Voll3r mentioned, Neon 2.24.1 actually has a feature that enables users to migrate your NEO/GAS from Neo Legacy to Neo N3, here is a video from NNT demonstrating it (https://www.youtube.com/watch?v=dyPwOvQuH9E).

Do notice that the UI might be slightly different from the video, as there have been changes since it was published, but overall, it should help.

Another important detail is that you need enough GAS to pay for the migration transaction, otherwise you will receive errors (it is free if you are migrating >= 10 NEO or >= 20 GAS.

Here is a useful note from Joe in our discord server about the migration _"It will take up to 72 hours but after that completes you should see the funds when you plug the Ledger in to Neon 3."_ 

A couple extra points regarding these topics:
- Neon 2.24.1 is enough to go through the migration by itself, with the feature described on that video.
- We are working on getting Ledger support for Neo Legacy on Neon3, but that has no connection to the Legacy > N3 migration process.
- As the Legacy > N3 migration feature is already supported on Neon 2.24.1, there are no plans at the moment to implement it on Neon3.

I hope this helps clarify things, please feel free to reach us for further questions, as well as join us on discord :)

**Comment by Voll3r** [comment]
Hi @thiagocbalducci. Thanks so much for the additional info. I was not aware that Neon 2.24.1 was still available. I just downloaded it (https://github.com/CityOfZion/neon-wallet) and am sure happy to be able to access my old funds again!

The migration function did not seem to work, but I did manage to transfer my old NEO to my Binance account without any problem.

Thanks!

**Comment by thiagocbalducci** [comment]
@Voll3r happy to help!

Could you share some details on what happened when you tried to migrate from legacy to N3?

About Neon 2.x, it will continue to be available in parallel to Neon 3.x, there is no schedule in place yet to formally discontinue it (the migration feature being one of the reasons, we don't want to lock users out of it).

**Comment by Voll3r** [comment]
Sure. I went to the migration tab here:

![image](https://github.com/user-attachments/assets/9d51f3ef-496b-488c-8660-a8248d9bb4b2)

Basically the animation didn't stop moving. There was no indication what specifically was happening and there was nothing shown on the ledger app either.

I waited for a bit, perhaps a few minutes. I did try going back and forth between tabs a few times, but as that didn't change anything I just decided to take the Binance route.

**Comment by gromakovsky** [comment]
@Voll3r @thiagocbalducci thank you for your help! I saw the Youtube video about the migration, but I didn't know Neon 2.anything was still available. I checked the releases in this repo, but found only 3.*. I've just installed 2.24.1 and it shoed my old reserves of NEO. The migration didn't work, I've got the same behavior as @Voll3r. Fortunately, I have a Binance account too, so I went this route and it worked, it cost me only 0.05 GAS to withdraw GAS (NEO withdrawal was free).

**Comment by fabwa** [comment]
The latest version helped to some degree. It now selects some default wallet. How do i import other wallets from the same Ledger? (The equivalent of this one in 2.x) 
<img width="565" alt="Screenshot 2024-10-22 at 09 19 05" src="https://github.com/user-attachments/assets/db11a833-ef75-499d-a5e7-b8c4cc7df13e">

cc @thiagocbalducci

**Comment by thiagocbalducci** [comment]
Hi @fabwa! The upcoming 3.2 will have this feature :)

This release is in an advanced testing stage already

**Comment by Trinh-at-NewsGroupDigital** [comment]
> Hi @thiagocbalducci. Thanks so much for the additional info. I was not aware that Neon 2.24.1 was still available. I just downloaded it (https://github.com/CityOfZion/neon-wallet) and am sure happy to be able to access my old funds again!
> 
> The migration function did not seem to work, but I did manage to transfer my old NEO to my Binance account without any problem.
> 
> Thanks!

@Voll3r 
I've been having the same problems using my Ledger Nano S and getting access to the address on the Neo Legacy chain using the Ledger as the private key. I found this thread and saw that you called out not being able to import a legacy walleg from Ledger but saw that you found using 2.24.1 worked to move your assets... was wondering if there was somewhere I can get the 2.24.1 binary from or do I have to figure out how to build the project myself? When I open my legacy Neo app on Ledger to log into NEON wallet it gives me two accounts with N3 addresses rather than allowing me to find the Legacy "A" starting address that I created back in 2021 using Ledger.

Would love to see NEON support importing Legacy NEO Ledger wallets 😄

**Comment by Voll3r** [comment]
Hi @Trinh-at-NewsGroupDigital,

yes, the binaries are available from the version 2 repository. Here's a direct link to the download page: https://github.com/CityOfZion/neon-wallet/releases/tag/v2.24.1
